src/js/structs/Node.js

```diff
@@ -39,8 +39,13 @@ class Node extends Item {
   }
 
   getUsageStats(resource) {
-    const total = this.get("resources")[resource];
-    const value = this.get("used_resources")[resource];
+    const total = (this.get("resources") || {})[resource];
+    const value = (this.get("used_resources") || {})[resource];
+
+    if (total === undefined || value === undefined) {
```

Comment:
Are we sure it can't be `null`? I think typeof checks might give the reader more context and us more security. Like 
```js 
typeof total === "number"
```

Reply:
You are right, I assumed that because it is an empty hash or a value, it can't never be null. But this would be better code.

Reply:
Also I think I have read somewhere in my life that JS never puts null for you, you have to put null there yourself.